### DISCUSSION TITLE
What is the maximum amount of unittest case data supported?

### DISCUSSION THREAD

**Discussion by xushaoxiao** [allure-framework/allure2#1529] [Questions & Support] [answered]
Hi forks,

I'm a backend engineer, currently i use allure to integration my test case result data by `allure serve -p 8080 ${directory}`.
When the specified directory contains more than 800,000 unittest cases, the allure tool cannot complete the generated work.

thx.

![image](https://user-images.githubusercontent.com/83151234/144781608-4df30567-9cb1-465c-b4a8-4943edaf70fa.png)

![image](https://user-images.githubusercontent.com/83151234/144781683-5ce5cc89-508c-48fe-822e-369b0d660453.png)

**Comment by baev** [comment, marked as answer]
allure loads all the results into RAM during report generation. And it may take it a while. 

But the main issue would be the generated report itself, a single DOM will be created for list of all the results, and I doubt any  browser can handle the DOM of 6GB.

I have no issues generating report of 300k junit.xml test results, the report laggy but it works.

**Comment by xushaoxiao** [comment]
can `allure serve` command specified RAM param?

**Comment by xushaoxiao** [comment]
in my issue, i used json and text files
